### PR TITLE
Add StackExchange integration

### PR DESCRIPTION
**Pull request by raphink**
This adds StackExchange integration. Two modes are supported:

* user reputation: you need to specify the site name (e.g. stackoverflow, superuser, tex, etc.) and the user id.
* questions by tag: you need to supply a site name and tag name (e.g. 'augeas').

### CODE REVIEW

**Comment by espadrine** [comment]
Thanks for contributing!

I believe showing the `display_name` on the left of the reputation badge would be more appropriate.

Similarly, I believe showing the tag on the left of the tag badge would make more sense.

(I notice we could similarly have `[ bronze | 133 ]` & cie badges.)

File: server.js [hunk 1]
```diff
@@ -3082,6 +3083,52 @@ cache(function(data, match, sendBadge, request) {
...
+  var options = {
+    method: 'GET',
+    uri: 'https://api.stackexchange.com/2.2/'+path+'?site='+site,
+    encoding: null
+  }
+  var badgeData = getBadgeData(site, data);
+  request(options, function (err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(badgeData, format);
```
Comment (espadrine) [inline server.js]: There should be a return statement.

File: server.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ var https = require('https');
 var domain = require('domain');
 var request = require('request');
 var fs = require('fs');
+var zlib = require('zlib');
```
Comment (espadrine) [inline server.js]: Remove this line (see below).

File: server.js [hunk 3]
```diff
@@ -3082,6 +3083,52 @@ cache(function(data, match, sendBadge, request) {
...
+  var path;
+  if (info === 'r') {
+    path = 'users/'+item;
+  } else if (info === 't') {
+    path = 'tags/'+item+'/info';
+  }
+  var options = {
+    method: 'GET',
+    uri: 'https://api.stackexchange.com/2.2/'+path+'?site='+site,
+    encoding: null
```
Comment (espadrine) [inline server.js]: Replace this line with `gzip: true`.

Change the required version of request from `"~2.34.0"` to `"~2.54.0"` in `package.json`. Do `npm install request`.

File: server.js [hunk 4]
```diff
@@ -3082,6 +3083,52 @@ cache(function(data, match, sendBadge, request) {
...
+  }
+  var badgeData = getBadgeData(site, data);
+  request(options, function (err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(badgeData, format);
+    }
+    zlib.gunzip(buffer, function (err, body) {
```
Comment (espadrine) [inline server.js]: Remove the zlib call; use buffer instead of body.

**Comment by larsbrinkhoff** [comment]
@raphink, this seems pretty cool.

**Comment by espadrine** [comment]
@larsbrinkhoff You can probably take over the PR if you want.

**Comment by larsbrinkhoff** [comment]
Can do.

Is there a way to transfer ownership of a PR?  Change from `raphink:dev/stackexchange` to my branch?

**Comment by raphink** [comment]
@larsbrinkhoff if you don't want to open a new PR, you can file a PR on my branch and I'll merge it. Otherwise, just open a new PR based on this. I don't really have time to fix this unfortunately.

**Comment by larsbrinkhoff** [comment]
I have cloned your branch: https://github.com/larsbrinkhoff/shields/commits/stackexchange

I will post a new pull request when I have made the requested fixes.  You can close this PR any time you like.

File: server.js (hunk 3 above)
Comment (larsbrinkhoff) [inline server.js]: request is now at "~2.55.0".

**Comment by espadrine** [comment]
Merged as #600.

**Pull request closed without merging**